File: TaskManagementSystem/TaskManagement.Web/src/app/models/task.model.ts
export interface Task {

    id: number;
    title: string;
    description?: string;
    statusId: number;
    statusName: string;
    createdByUserId: number;
    createdByUsername: string;
}

export interface CreateTask {

    title: string;
    description?: string;
    statusId: number;
}

export interface UpdateTask {
    
    id: number;
    title: string;
    description?: string;
    statusId: number;
}

File: TaskManagementSystem/TaskManagement.Domain/Models/TaskModel.cs
namespace TaskManagement.Domain.Models
{
    public class TaskModel
    {
        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public bool IsDeleted { get; set; }
        public int StatusId { get; set; }
        public string StatusName { get; set; } = null!;
        public int UserId { get; set; }
    }
}